spec/requests/api/v1/provider_events_spec.rb

```diff
@@ -8,6 +8,15 @@
     let(:token) { LeadProviderApiToken.create_with_random_token!(lead_provider: lead_provider) }
     let(:bearer_token) { "Bearer #{token}" }
     let(:payload) { create(:early_career_teacher_profile) }
+    let(:delivery_partner) { create(:delivery_partner) }
+    let!(:school_cohort) { create(:school_cohort, school: payload.school, cohort: payload.cohort) }
+    let!(:partnership) do
+      create(:partnership,
+             school: payload.school,
+             lead_provider: lead_provider,
+             cohort: payload.cohort,
+             delivery_partner: delivery_partner)
+    end
```

Comment:
So we added a bunch of setup, but we haven't added any tests. I would expect a unit test like "when lead providers don't match, recording participant event raises an error".

Probably not in this file, I imagine `record_participant_event` could use its own test suite to demonstrate and test those edge cases.